### PR TITLE
add profile on User model and Serializer

### PR DESCRIPTION
**Pull request by RignonNoel**
| Q                                                      | R
| ------------------------------------------ | -------------------------------------------
| Type of contribution ?                      | Enhancement
| Tickets (_issues_) concerned               | None

---

##### What have you changed ?
Add some profile information on user

##### How did you change it ?
 - Auto create profile in serializer
 - Link items of profile in the user serializer
 - Modify test
 - Add a model `Profile` with a OneToOne relation with user

Verification :
 -  [x] My branch name respect the standard defined in `CONTRIBUTING.md`
 -  [ ] This Pull-Request fully meets the requirements defined in the issue
 -  [x] I added or modified the attached tests

### CODE REVIEW

File: source/apiVolontaria/apiVolontaria/serializers.py [hunk 1]
```diff
@@ -93,13 +95,28 @@ class Meta:
...
+        required=False,
+    )
+    mobile = serializers.CharField(
+        source='profile.mobile',
+        required=False,
+    )
+
     def create(self, validated_data):
-        user = User(**validated_data)
+        profile_data = validated_data.pop('profile')
```
Comment (RignonNoel) [inline source/apiVolontaria/apiVolontaria/serializers.py]: Add a condition if no key `profile`

File: source/apiVolontaria/apiVolontaria/models.py [hunk 2]
```diff
@@ -86,3 +86,23 @@ def expire(self):
...
+        User,
+        on_delete=models.CASCADE,
+        related_name='profile',
+    )
+    phone = models.CharField(
+        verbose_name="Phone number",
+        max_length=15,
+        null=True,
+        blank=True,
+    )
```
Comment (DarkFranX) [inline source/apiVolontaria/apiVolontaria/models.py]: Aren't there any predefined phone number format? We should probably use a regular expression here to be sure a correct number is sent to the DB. Have a look at this [Stack Overflow answer](https://stackoverflow.com/a/19131360) giving some guidelines for a fail-safe phone number field.

File: source/apiVolontaria/apiVolontaria/models.py [hunk 3]
```diff
@@ -86,3 +86,23 @@ def expire(self):
...
+    mobile = models.CharField(
+        verbose_name="Mobile number",
+        max_length=15,
+        null=True,
+        blank=True,
+    )
```
Comment (DarkFranX) [inline source/apiVolontaria/apiVolontaria/models.py]: Same here

File: source/apiVolontaria/apiVolontaria/serializers.py [hunk 4]
```diff
@@ -93,13 +95,31 @@ class Meta:
...
+    mobile = serializers.CharField(
+        source='profile.mobile',
+        required=False,
+    )
+
     def create(self, validated_data):
-        user = User(**validated_data)
+        profile_data = None
+        if 'profile' in validated_data.keys():
+            profile_data = validated_data.pop('profile')
```
Comment (DarkFranX) [inline source/apiVolontaria/apiVolontaria/serializers.py]: This is cool. Nicely done!

**Pull request merged**